app/controllers/statuses_controller.rb

```diff
@@ -66,7 +66,12 @@ def set_link_headers
 
   def set_status
     @status = @account.statuses.find(params[:id])
-    authorize @status, :show?
+
+    if request.authorization.present? && request.authorization.match(/^Bearer /i)
+      raise Mastodon::NotPermittedError unless @status.capability_tokens.find_by(token: request.authorization.gsub(/^Bearer /i, ''))
+    else
+      authorize @status, :show?
+    end
```

Comment:
Added `Authorization` to `Vary` headers.

I used `request.authorization` instead of `request.headers['Authorization']` because I saw that doorkeeper does that and wanted to be consistent with that.

Reply:
I get that, I'm just worried that it may possibly lead to security issues if an implementation ends up somehow using something else than the `Authorization` header that `request.authorization` will use. I know for sure it accepts variants of `Authorization` and I wonder if Devise or something doesn't make it also support other things like the `bearer_token` field in form-data POST. This would mean such an implementation would possibly enable caching of something private without an `Authorization` header.

Reply:
The code for `request.authorization` is:

```ruby
def authorization
  get_header("HTTP_AUTHORIZATION")   ||
  get_header("X-HTTP_AUTHORIZATION") ||
  get_header("X_HTTP_AUTHORIZATION") ||
  get_header("REDIRECT_X_HTTP_AUTHORIZATION")
end
```

Admittedly, I don't understand exactly when `X-HTTP_AUTHORIZATION` would be set vs `X_HTTP_AUTHORIZATION` etc, but it definitely does not touch form-data. I am *guessing* that these header variants are the different ways the `Authorization` header may be forwarded by Apache and other proxies but not knowing the exact reasons does make me uncomfortable.

Perhaps I should use `request.headers['Authorization']` directly.